CriteoPublisherSdk/Sources/Feedback/CR_FeedbackController.m

```diff
@@ -111,18 +113,36 @@ - (void)onBidConsumed:(CR_CdbBid *)consumedBid {
 }
 
 - (void)sendFeedbackBatch {
-  NSArray<CR_FeedbackMessage *> *feedbackMessages = [self.feedbackStorage popMessagesToSend];
-  if (feedbackMessages.count == 0) {
+  NSArray<CR_FeedbackMessage *> *messages = [self.feedbackStorage popMessagesToSend];
+  if (messages.count == 0) {
     return;
   }
 
+  NSDictionary<NSNumber *, NSArray<CR_FeedbackMessage *> *> *messagesByProfileId =
+      [messages cr_groupByKey:^NSNumber *(CR_FeedbackMessage *message) {
+        return message.profileId ?: @(CR_IntegrationFallback);
+      }];
+  [messagesByProfileId
+      enumerateKeysAndObjectsUsingBlock:^(NSNumber *profileId,
+                                          NSArray<CR_FeedbackMessage *> *messages, BOOL *stop) {
```

Comment:
this messages variable is shadowing the one above. What do you think about using a more meaningful name for this one ?